sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -252,16 +252,30 @@ class RocksDB(
     }
   }
 
+  /**
+   * Check whether the column family name is for internal column families.
+   * @param cfName - column family name
+   * @return - true if the column family is for internal use, false otherwise
+   */
+  private def checkInternalColumnFamilies(cfName: String): Boolean = cfName.charAt(0) == '_'
+
   /**
    * Create RocksDB column family, if not created already
    */
-  def createColFamilyIfAbsent(colFamilyName: String): Unit = {
-    if (colFamilyName == StateStore.DEFAULT_COL_FAMILY_NAME) {
+  def createColFamilyIfAbsent(colFamilyName: String, isInternal: Boolean = false): Unit = {
+    // Remove leading and trailing whitespaces
+    val cfName = colFamilyName.trim
+
+    if (cfName == StateStore.DEFAULT_COL_FAMILY_NAME) {
       throw new SparkUnsupportedOperationException(
         errorClass = "_LEGACY_ERROR_TEMP_3197",
         messageParameters = Map("colFamilyName" -> colFamilyName).toMap)
     }
 
+    if (!isInternal && cfName.charAt(0) == '_') {
```

Comment:
We should add a check that the cfName is not empty string, and throw an error. [Right now we will end in IndexOutofBoundsException]

Reply:
Done